openrtb_ext/floors.go

```diff
@@ -0,0 +1,114 @@
+package openrtb_ext
+
+// Defines strings for FetchStatus
+const (
+	FetchSuccess    = "success"
+	FetchTimeout    = "timeout"
+	FetchError      = "error"
+	FetchInprogress = "inprogress"
+	FetchNone       = "none"
+)
+
+// Defines strings for PriceFloorLocation
+const (
+	NoDataLocation  = "noData"
+	RequestLocation = "request"
+	FetchLocation   = "fetch"
+)
+
+// PriceFloorRules defines the contract for bidrequest.ext.prebid.floors
+type PriceFloorRules struct {
+	FloorMin           float64                `json:"floormin,omitempty"`
+	FloorMinCur        string                 `json:"floormincur,omitempty"`
+	SkipRate           int                    `json:"skiprate,omitempty"`
+	Location           *PriceFloorEndpoint    `json:"floorendpoint,omitempty"`
+	Data               *PriceFloorData        `json:"data,omitempty"`
+	Enforcement        *PriceFloorEnforcement `json:"enforcement,omitempty"`
+	Enabled            *bool                  `json:"enabled,omitempty"`
+	Skipped            *bool                  `json:"skipped,omitempty"`
+	FloorProvider      string                 `json:"floorprovider,omitempty"`
+	FetchStatus        string                 `json:"fetchstatus,omitempty"`
+	PriceFloorLocation string                 `json:"location,omitempty"`
```

Comment:
Nitpick: Do you think there's a better json name that could be used here? It being just `location` makes me think about the actual member `Location` of the `PriceFloorRules` struct. Just wanted to bring this up in case you can think of one

Reply:
The json field name `location` is added just to align analytics adaptor price floors field name 
https://docs.google.com/document/d/1YAt00RfQaU2Ugciqw1LnirLgjyUz9ibfyjfxWUHR_P0/edit#heading=h.z87hlqsz6yb